src/abstract-classes/browser-plugin.js

```diff
@@ -28,17 +31,26 @@ class BrowserPlugin {
      *  Automation libraries configure proxies differently. This helper allows you
      *  to set a proxy URL without worrying about specific implementations.
      *  It also allows you use an authenticated proxy without extra code.
+     * @property {boolean} [useIncognitoPages=false]
+     *  By default pages share the same browser context.
+     *  If set to true each page uses its own context that is destroyed once the page is closed or crashes.
+     * @property {object} [userDataDir]
+     *  Path to a User Data Directory, which stores browser session data like cookies and local storage.
      */
     constructor(library, options = {}) {
         const {
             launchOptions = {},
             proxyUrl,
+            useIncognitoPages = false,
+            userDataDir = path.join(os.tmpdir(), nanoid()),
```

Comment:
The `userDataDir` should change for every launched browser. This would keep it the same for all browsers launched with the same plugin.

We want to change cookies when we're changing proxies.